cmd/kube-controller-manager/app/core.go

```diff
@@ -504,3 +525,21 @@ func processCIDRs(cidrsList string) ([]*net.IPNet, bool, error) {
 
 	return cidrs, dualstack, nil
 }
+
+// processNodeCIDRMasks is a helper function that works on comma separated cidr masks and returns
+// a list of int mask sizes
+// error if failed to parse any of the mask size
+func processNodeCIDRMasks(nodeCIDRMaskList string) ([]int, error) {
+	nodeCIDRMaskSplit := strings.Split(strings.TrimSpace(nodeCIDRMaskList), ",")
+	var nodeCIDRMasks []int
+	if !(len(nodeCIDRMaskSplit) == 1 && nodeCIDRMaskSplit[0] == "") {
```

Comment:
This chunk of code is pretty hard to read, can be easier to read like this:
```go
if len(nodeCIDRMaskSplit) == 1 {
     return nodeCIDRMasks
}

for _, nodeCIDRMask := range nodeCIDRMaskSplit {
    intMask, err := strconv.Atoi(nodeCIDRMask)
    if err != nil {
        return nil, err
    }
    
    nodeCIDRMasks = append(nodeCIDRMasks, intMask)
}

return nodeCIDRMasks, nil
```

Reply:
should validate the node cidr masks value prior to calling this as well

Reply:
Thanks @andrewsykim. I've updated it so it can be easier to read. The validation to ensure conversion to int is done here. Also, the check to see if the node cidr mask size is not less than cluster cidr mask size is done here - https://github.com/kubernetes/kubernetes/pull/79993/files#diff-6bbac02bd63acca6e05007f7c054396cR115-R118.

Reply:
> The validation to ensure conversion to int is done here

Not for the len(nodeCIDRMasks) == 1 case I think?

Reply:
The check to see if the size of slice after `strings.Split` https://github.com/kubernetes/kubernetes/pull/79993/files#diff-0ca8b7870d367d5fc309a2e737a6eed2R536-R538 is done because if the nodeCIDRMaskList is `""` it results in an empty slice with len `1`. In that case we return the empty `nodeCIDRMasks`. Otherwise we go into the for loop and do the validation for each mask defined.